chainstate/src/detail/tests/mod.rs

```diff
@@ -47,6 +49,19 @@ const ERR_STORAGE_FAIL: &str = "Storage failure";
 const ERR_CREATE_BLOCK_FAIL: &str = "Creating block caused fail";
 const ERR_CREATE_TX_FAIL: &str = "Creating tx caused fail";
 
+#[must_use]
+fn make_seedable_rng(seed_opt: Option<&str>) -> impl Rng {
```

Comment:
Minor suggestion here. I think there is room for some improvement in how the seed is passed in. Instead of passing in an `Option`, introduce a new type, called say `Seed`, which wraps `[u8; 32]` and provides constructors for various use cases:
* `Seed::nondeterministic()`
* `Seed::these_exact_bytes_i_tell_you(seed: [u8; 32])`
* `Seed::from_str(seed_hex_str: &str)`

I do not insist on these particular names btw

Reply:
I think that 32-byte seed is an overkill... we've always lived with 32-bit seeds... and 64-bit is fine. But seems like Gosha is using Chacha, which is an overkill. Fine, that's all OK.

But, I'm totally with Lukas here. I came here to make a similar comment. The generation of the seed should not be manually handled by the called. Some wrapper type that makes this automatically done is a good idea.